### ISSUE TITLE
VoVGSCSP中的res层未使用

### ISSUE THREAD

**Issue by chuanqi1997** [alanli1997/slim-neck-by-gsconv#39]
大佬您好，我在自己的研究中使用了VoVGSCSPC，但是发现了res 层未使用，但是在训练时会占用模型的参数，并且去掉self.res 的定义，模型的精度会降低。
请问您是故意把res层放到这里的，还是不小心。
`class VoVGSCSP(nn.Module):
    # VoVGSCSP module with GSBottleneck
    def __init__(self, c1, c2, n=1, shortcut=True, g=1, e=0.5):
        super().__init__()
        c_ = int(c2 * e)  # hidden channels
        self.cv1 = Conv(c1, c_, 1, 1)
        self.cv2 = Conv(c1, c_, 1, 1)
        # self.gc1 = GSConv(c_, c_, 1, 1)
        # self.gc2 = GSConv(c_, c_, 1, 1)
        self.gsb = GSBottleneck(c_, c_, 1, 1)
        self.res = Conv(c_, c_, 3, 1, act=False)
        self.cv3 = Conv(2*c_, c2, 1)  #

    def forward(self, x):

        x1 = self.gsb(self.cv1(x))
        y = self.cv2(x)
        return self.cv3(torch.cat((y, x1), dim=1))`

**Comment by alanli1997** [comment]
> 大佬您好，我在自己的研究中使用了VoVGSCSPC，但是发现了res 层未使用，但是在训练时会占用模型的参数，并且去掉self.res 的定义，模型的精度会降低。 请问您是故意把res层放到这里的，还是不小心。

你好。这是个有趣的现象，res确实没有加入到模型前馈。这个项目时间比较久了，我觉得我应该是做消融实验后忘记了注释这一个res连接。消融实验里我记得是没有额外res的卷积块性能更好。理论上注释掉res不会对模型精度造成任何影响。难道是影响了模型初始化所以导致精度变化🤔？这个可能需要你来进一步探索。

额外的，gsconv是三年前的工作了，我们推出了和即将推出新工作，可能对你有用：
[https://github.com/alanli1997/rethinking-fpn](https://github.com/alanli1997/rethinking-fpn)
[https://github.com/alanli1997/slvm](https://github.com/alanli1997/slvm)
